Update OPAL to Scala 2.13

svenkeidel opened the pull request:
Notable changes:
- Replace unicode arrows (`⇒`,`→`,`←`) by ascii arrows (`=>`,`->`,`<-`)
- Replace `Traversable` by `scala.collection.Iterable`
- Replace `TraversableOnce` by `scala.collection.IterableOnce`
- Changes to OPAL's custom data structures:
  - Replace `RefIterator` by `scala.collection.iterator`
  - Replace `Chain` by `scala.collection.immutable.List`
  - Replace `RefArray` and `IntArray` by `scala.collection.immutable.ArraySeq`
  - Replace `RefArrayStack` by `scala.collection.immutable.Stack`
  - Replace `RefAppendChain` by `scala.collection.mutable.ArrayDeque`
  - Replace `RefArrayBuffer` by `scala.collection.mutable.ArrayBuffer`

TorunR reviewed (approved):
Wollen wir durch eigenes Tracking ersetzen? Nicht zwingenderweise Google ...

errt commented:
Also ich brauche jetzt auch kein Google Analytics. Grundsätzlich sind so Statistiken ganz nett anzuschauen, gerade wenn man irgendwie nen Vortrag oder so hatte und schauen will, was das für Auswirkungen hatte. Aber wirklich brauchen tun wir das eigentlich nicht. Wenn es also einfach geht und noch jemand Nutzen darin sieht, können wir das gerne machen, aber Priorität hat es für mich jetzt nicht.

svenkeidel committed:
change scala version 2.13 -> 2.12, fix compiler flags

svenkeidel committed:
Reformat unicode arrows to ascii arrows

Unicode arrows are not supported anymore since Scala 2.13

svenkeidel committed:
Fix: auto-application to () is deprecated

svenkeidel committed:
replace ConstArray by immutable.ArraySeq

svenkeidel committed:
Fix deprecated syntax

- Multi-argument infix notation is deprecated (e.g., x + (y,z))
- Constructor definitions need to use an `=` instead of curly braces

svenkeidel committed:
Remove RefArray and IntArray

svenkeidel committed:
Update UIDSet and IntArrayStack

svenkeidel committed:
Replace RefArrayStack by scala.collections.mutable.Stack

svenkeidel committed:
Remove OpenHashMap from Map benchmark since it is deprecated in 2.13

svenkeidel committed:
Replace Chain by List

svenkeidel committed:
Use scala.jdk.CollectionConverters

svenkeidel committed:
Fix .iterator errors in AbstractGraph

svenkeidel committed:
Fix various errors in common package

svenkeidel committed:
Apply reformatting

svenkeidel committed:
Replace unicode ← by <-

svenkeidel committed:
Replace RefIterator with scala.collections.Iterator

svenkeidel committed:
Replace Traversable by Iterable and RefArrayBuffer by ArrayBuffer

svenkeidel committed:
More changes to update collections

svenkeidel committed:
Fix header in UIDSet

svenkeidel committed:
Enable assertions for development

errt committed:
Replace 'Empty and more unicode arrows

errt committed:
Move assertions option to the end where it's easier to find

errt committed:
Fix type error

svenkeidel committed:
Add entity and property type parameters to EOptionP unapply methods

svenkeidel committed:
Replace some occurences of IntIterator with Iterator[Int]

svenkeidel committed:
Fix compile errors

errt committed:
Fix incorrect fix

svenkeidel committed:
Fix lots of tests

svenkeidel committed:
Fix unexhaustive patterns in test

svenkeidel committed:
Change Traversables to Iterable

errt committed:
Formatting

errt committed:
Can't use string interpolation here as it confuses the formatter

errt committed:
Renaming to avoid shadowing of inner classes

errt committed:
Use string interpolation

svenkeidel committed:
fix non-exhaustive pattern matches

svenkeidel committed:
use nightly version of Scala 2.13.9

svenkeidel committed:
Turn some vars to vals

svenkeidel committed:
Suppress switch warnings

svenkeidel committed:
Remove further uses of RefArray

svenkeidel committed:
Fix errors in Project

svenkeidel committed:
Fix all errors in directory OPAL

svenkeidel committed:
Fix UIDSet

svenkeidel committed:
Fix UIDSet's map and union

svenkeidel committed:
change mutable sets in PropertyStoreTest

svenkeidel committed:
move Constant_Pool_Entry class tag to super trait

svenkeidel committed:
Fix class hierarchy

svenkeidel committed:
Fix errors changing `UIDSet.++` to `++=`

svenkeidel committed:
Fix early initialization problem in ClassFileReaderConfiguration

svenkeidel committed:
Fix more classTag problems

svenkeidel committed:
Fix problem with TABLESWITCH and ClassFile

svenkeidel committed:
Fix bug in InvokedynamicRewriting

svenkeidel committed:
Fix missing class tag in Record_attributeBinding

svenkeidel committed:
Formatting

svenkeidel committed:
Fix join operator

svenkeidel committed:
Fix ordering of successors in TACAI cfg

svenkeidel committed:
Fix bytecode assembler tests

svenkeidel committed:
Fix architecture validation tests

svenkeidel committed:
Fix sbt template

svenkeidel committed:
Fork process to pick up javaOptions

svenkeidel committed:
Fix OPAL_DEVELOP compile errors

svenkeidel committed:
Fix points to analysis

errt committed:
Remove debug code

errt committed:
Add warning configuration to scalac.options.local.template

svenkeidel committed:
Fix scalariform and unidoc errors

svenkeidel committed:
Revert fork options in build.sbt

svenkeidel committed:
Fix broken imports in tests

svenkeidel committed:
Fix "-encoding" commandline option

svenkeidel committed:
Do not publish test artifacts

errt committed:
Format imports

errt reviewed (dismissed):
LGTM

errt committed:
Fix broken import

errt committed:
Fix more broken imports

errt committed:
Yet another broken import...
